### ISSUE TITLE
Handling of SD card missing when accessing saved geocache

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6451]
##### Detailed steps causing the problem:
- Have GeocacheData moved to external SD card
- Close and kill c:geo
- Remove SD card from device
- Start c:geo
- Open saved caches

##### Actual behavior after performing these steps:
Crash

##### Expected behavior after performing these steps:
Either we should fallback to internal dir if the SD card was removed (see #6450) or at least missing additional geocache data should not lead to a application crash.

##### Version of c:geo used:
2017.03.25-NB

##### Is the problem reproducible for you?
Yes

##### System information:
Android 5.1.1
Samsung Galaxy Note 10.1 Edition 2014 (SD removeable without power down)

##### Other comments and remarks:

Log:
```
12:15:58.833 Warning cgeo 27275  [RxComputationThreadPool-3] HtmlImage.loadImageFromStorage
12:15:58.833 Warning cgeo 27275  java.lang.NullPointerException: Attempt to invoke virtual method 'java.io.File java.io.File.getCanonicalFile()' on a null object reference
12:15:58.833 Warning cgeo 27275  	at android.os.Environment.getStorageVolume(Environment.java:900)
12:15:58.833 Warning cgeo 27275  	at android.os.Environment.getExternalStorageState(Environment.java:758)
12:15:58.833 Warning cgeo 27275  	at android.os.Environment.getStorageState(Environment.java:744)
12:15:58.833 Warning cgeo 27275  	at android.support.v4.os.EnvironmentCompatKitKat.getStorageState(EnvironmentCompatKitKat.java:25)
12:15:58.833 Warning cgeo 27275  	at android.support.v4.os.EnvironmentCompat.getStorageState(EnvironmentCompat.java:58)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.storage.LocalStorage.getAvailableExternalPrivateCgeoDirectories(LocalStorage.java:82)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.storage.LocalStorage.getExternalPrivateCgeoDirectory(LocalStorage.java:98)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.storage.LocalStorage.getGeocacheDataDirectory(LocalStorage.java:281)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.storage.LocalStorage.getGeocacheDataDirectory(LocalStorage.java:156)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.storage.LocalStorage.getGeocacheDataFile(LocalStorage.java:174)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.network.HtmlImage.loadImageFromStorage(HtmlImage.java:366)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.network.HtmlImage.access$600(HtmlImage.java:55)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.network.HtmlImage$3.loadFromDisk(HtmlImage.java:253)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.network.HtmlImage$3.access$300(HtmlImage.java:216)
12:15:58.833 Warning cgeo 27275  	at cgeo.geocaching.network.HtmlImage$3$2.run(HtmlImage.java:231)
12:15:58.833 Warning cgeo 27275  	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
12:15:58.833 Warning cgeo 27275  	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
12:15:58.833 Warning cgeo 27275  	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
12:15:58.833 Warning cgeo 27275  	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
12:15:58.833 Warning cgeo 27275  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
12:15:58.833 Warning cgeo 27275  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
12:15:58.833 Warning cgeo 27275  	at java.lang.Thread.run(Thread.java:818)
``´

**Comment by pstorch** [comment]
should be fixed by #6453

**Comment by Lineflyer** [comment]
Assigning for fieldtest

**Comment by Lineflyer** [comment]
Confirmed fixed with 2017.04.23-RC

- Absence of SD is recognized on startup
- On accessing a saved cache the additional data (pictures) are redownloaded and stored on internal storage.
